addons/mail/static/src/widgets/form_renderer/form_renderer_tests.js

```diff
@@ -853,6 +853,82 @@ QUnit.test('Form view not scrolled when switching record', async function (asser
     );
 });
 
+QUnit.test('chatter remove attachment', async function (assert) {
+    assert.expect(3);
+
+    this.data['res.partner'].records.push(
+        { display_name: "first partner", id: 11 },
+        { display_name: "second partner", id: 12 }
+    );
+
+    this.data['ir.attachment'].records.push(
+        { display_name: "A1", id: 11, name: "AAAA", mimetype:"text.txt", res_id:11, res_model:'res.partner' },
+        { display_name: "A2", id: 12, name: "BBBB", mimetype:"text.txt", res_id:11, res_model:'res.partner' },
+        { display_name: "A3", id: 13, name: "CCCC", mimetype:"text.txt", res_id:12, res_model:'res.partner' },
+    );
+    await this.createView({
+        data: this.data,
+        hasView: true,
+        // View params
+        View: FormView,
+        model: 'res.partner',
+        res_id: 11,
+        viewOptions: {
+            ids: [11, 12],
+            index: 0,
+        },
+        arch: `
+            <form string="Partners">
+                <sheet>
+                    <field name="name"/>
+                </sheet>
+                <div class="oe_chatter">
+                    <field name="message_ids"/>
+                </div>
+            </form>
+        `,
+    });
+
+    assert.containsNone(
+        document.body,
+        '.o_AttachmentList_attachment',
+        "there should be no attachments displayed"
+    );
+
+    await afterNextRender(() =>
+        document.querySelector('.o_ChatterTopbar_buttonAttachments').click()
+    );
+
+    assert.containsN(
+        document.body,
+        '.o_AttachmentList_attachment',
+        2,
+        "There should be two attachments displayed"
+    );
+
+    await afterNextRender(async () => {
+        document.querySelector('.o_pager_next').click();
+    });
+
+    // Remove one attachment of "first partner". 
+    this.data['ir.attachment'].records.find(a => a.id === 11).res_id = 12
+
+    await afterNextRender(async () => {
+        document.querySelector('.o_pager_previous').click();
+    });
+
+    await afterNextRender(() =>
+        document.querySelector('.o_ChatterTopbar_buttonAttachments').click()
+    );
+
+    // Check if the attachment is remove on the view.
+    assert.containsOnce(
+        document.body,
+        '.o_AttachmentList_attachment',
+        "There should be one attachments displayed"
+    );
```

Comment:
Test could be shortened to just 2 user interactions and 2 assertions:

- **(assertion 1)** There are 2 attachments linked to "Partner1" initially.
- **(user interaction 1)** Move to Partner2 from pager
- **(simulated removable of one attachment on "Partner1")**
- **(user interaction 2)** Move back to Partner1 from pager
- **(assertion 2)** There is 1 attachment linked to "Partner2" after one attachment has been removed.